### PR TITLE
Filter Optimization & Image Lazy Loading

### PR DESCRIPTION
**Pull request by QuatZo**
* Added Lazy loading for all images
* Cleaned up filter url (i.e. `name=&natural_stars=1,6&fusion_food=&skills__passive=&skills__aoe=&skills__cooltime=0,13&skills__hits=0,7&page=1&sort=` changed to `page=1`)
* Optimized Bestiary/Monsters Database calls

It also partially resolves given Feedback
> Whenever you click on bestiary, swarfarm load every monster before you can modify filter, making swarfam super slow especicaly on my phone, maybe, for the first loading, don't load monster and wait for the user to press apply to filter.

| Loading time [seconds]  | SWARFARM (before) | LOCAL (after) |
|---|---|-----|
| Bestiary  | 7  | 1  |  
| Monsters  | 6.5  | 3.5   |  
| Runes  | 10  | 8  |
| Artifacts | 1.5 | <0.5 |

### CODE REVIEW

File: herders/templates/herders/profile/monster_inventory/list.html [hunk 1]
```diff
@@ -66,7 +66,7 @@
                             <th>Notes</th>
                             <th data-sorter="false" data-columnSelector="false">Teams</th>
                             <th data-sorter="false">Labels</th>
-                            <th class="monster-info-icons-column" data-selector-name="Fusion Food"><img src="{{ img_url_prefix }}icons/fusion.png"  data-toggle="tooltip" data-placement="top" title="Used in monster fusion"/></th>
+                            <th class="monster-info-icons-column" data-selector-name="Fusion Food"><img src="{{ img_url_prefix }}icons/fusion.png" loading="lazy" data-toggle="tooltip" data-placement="top" title="Used in monster fusion" loading="lazy" /></th>
```
Comment (PeteAndersen) [inline herders/templates/herders/profile/monster_inventory/list.html]: `loading="lazy"` added twice

File: bestiary/filters.py [hunk 2]
```diff
@@ -100,29 +114,49 @@ def filter_skill_effects(self, queryset, name, value):
...
+                skills = skills.filter(hits_filter)
 
             if passive is not None:
                 skills = skills.filter(passive=passive)
 
             if aoe is not None:
                 skills = skills.filter(aoe=aoe)
 
+            # no skill filters
+            if skills_count == skills.count():
```
Comment (PeteAndersen) [inline bestiary/filters.py:157]: Do you have any numbers for this specific optimization? Just want to make sure that adding 2 `count()` calls is worth the time to determine if we can skip the `filter(skills_in=skills)`

File: herders/filters.py [hunk 3]
```diff
@@ -111,29 +125,49 @@ def filter_monster__skills__skill_effect__pk(self, queryset, name, value):
...
+                skills = skills.filter(hits_filter)
 
             if passive is not None:
                 skills = skills.filter(passive=passive)
 
             if aoe is not None:
                 skills = skills.filter(aoe=aoe)
 
+            # no skill filters
+            if skills_count == skills.count():
```
Comment (PeteAndersen) [inline herders/filters.py:168]: Just noting that same answer from monster filter optimization question applies here

File: bestiary/filters.py (hunk 2 above)
Comment (QuatZo) [reply bestiary/filters.py:157]: Checked only for Bestiary, I expect similar results in Profile Monsters.

* Empty filter (`page=1`):
  * With double `count` check: 51ms whole db call
  * Without `count` check: 225ms whole db call (5 queries), `WHERE` clause has subquery: `WHERE ("bestiary_monster"."obtainable" = true AND "bestiary_monster_skills"."skill_id" IN (SELECT
"bestiary_skill"."id"
FROM "bestiary_skill"))`

* Full skill filter (no records returned, `buffs=7&buffs=13&buffs=70&debuffs=22&debuffs=23&debuffs=31&other_effects=51&other_effects=56&skills__scaling_stats__pk=24&skills__scaling_stats__pk=26&skills__passive=False&skills__aoe=True&skills__cooltime=3%2C8&skills__hits=2%2C6&leader_skill__attribute=2&leader_skill__attribute=5&leader_skill__area=2&leader_skill__area=5&page=1`):
  * With double `count` check: 226ms whole db call
  * Without `count` check: 200ms whole db call (203ms)

* Normal skill filter (Bomb debuff, `debuffs=23&page=1`):
  * With double `count` check: 50ms whole db call
  * Without `count` check: 33ms whole db call

Looks like it's quite a big deal for empty filter, but whenever we want to filter by anything from `skills`, then it's small downgrade. Of course I did only like 3 rounds per test, so it's not a big sample size.

**Pull request merged**